Unique pallet

rohibhan opened the pull request:
Need a Unique pallet with the following functionalities.
- [x] Digest of the document/data
- [x] Authorisation ID support
- [x] Optional attribute - stream identifier
- [x] If this option is used, check the status and ownership of the identifier.
- [x] Add stream identifier link to the unique storage
- [x] Create a new identifier - use a new prefix.
- [x] Revoke / Restore functionality.

rohibhan committed:
Added Pallet Unique Structure

rohibhan committed:
Runtime entry for reading pallet

rohibhan committed:
Fixed missing comma in runtime file

roynrishingha committed:
Release CORD Arch x86 compiled binary from github action (#178)

- Upload the binary as a release asset
- User should be able to `wget` or `curl` the pre-compiled binary

rohibhan committed:
create_function_updated_uniques

roynrishingha committed:
Fix cord binary path (#180)

amarts commented:
Can the build errors be fixed in this?

rohibhan committed:
added revoke

rohibhan commented:
fixed errors and added revoke function

amarts commented on pallets/unique/Cargo.toml:
```diff
@@ -0,0 +1,74 @@
+cord-primitives = { package = "cord-primitives", path="../../primitives", default-features = false }
+cord-utilities = { package = "cord-utilities", path = "../../utilities", default-features = false }
+pallet-registry = {  default-features = false, path = '../registry' }
+pallet-schema = {  default-features = false, path = '../schema' }
```
This dependency is not required I guess. Similarly review every presence of 'Schema' in the PR.

amarts commented on pallets/unique/src/types.rs:
```diff
@@ -0,0 +1,92 @@
+///
+/// * `digest`: The hash of the stream.
+/// * `creator`: The account that created the stream.
+/// * `schema`: The schema identifier.
```
wrong doc, remove schema

amarts commented on primitives/src/curi.rs:
```diff
@@ -75,6 +77,8 @@ pub enum IdentifierType {
 	Authorization,
 	Schema,
 	Stream,
+	//TODO : added unique in identifire type
```
is it still a TODO?

also minor nitpick: 's/identifire/identifier/g'

rohibhan committed:
removed stream comments

smohan-dw committed:
chore: release-v0.9.43

smohan-dw committed:
chore: add sudo the benchmarks

smohan-dw committed:
test utils fix

amarts commented:
Plan to use it in one of the demo script by end of the month. Please focus on getting the tests passed, and lets get this merged.

rohibhan commented:
> Plan to use it in one of the demo script by end of the month. Please focus on getting the tests passed, and lets get this merged.

Sure plannig to complete it by end of this month
Thanks

roynrishingha committed:
Fix release path and branch

- Release will run on `develop` branch and with any other branch name begin with `release`.

roynrishingha committed:
Change telemetry URL to `telemetry.cord.network`.

rohibhan committed:
Added Storage,bounded vec to limit unique hash siz

roynrishingha committed:
Fix syntax error in Github Action. (#183)

roynrishingha committed:
Fix Github Action Release Error (#184)

amarts commented:
Can you rebase it on top of develop ?

rohibhan committed:
replace to_owned  to clone

rohibhan committed:
change dep version

rohibhan committed:
removed benach mark for now

rohibhan committed:
proper error handling

smohan-dw commented on pallets/unique/src/lib.rs:
```diff
@@ -0,0 +1,429 @@
+	#[pallet::storage]
+	#[pallet::getter(fn unique_identifiers)]
+	pub type UniqueIdentifiers<T> =
+		StorageMap<_, Blake2_128Concat, InputUniqueOf<T>, UniqueIdOf, OptionQuery>;
```
Storage here should be named as UniqueDigestEntries, as it's mapping the input to the identifier.

smohan-dw commented on pallets/unique/src/lib.rs:
```diff
@@ -0,0 +1,429 @@
+	/// It maps from a unique hash to an metadata (resolve from hash).
+	#[pallet::storage]
+	#[pallet::getter(fn unique_digest_entries)]
+	pub type UniqueDigestEntries<T> =
```
This one should be UniqueIdentifiers as it is mapping the Identifier to details.

smohan-dw commented on pallets/unique/src/lib.rs:
```diff
@@ -0,0 +1,429 @@
+		Create { identifier: UniqueIdOf, digest: InputUniqueOf<T>, author: UniqueCreatorIdOf<T> },
+		/// A unique identifier status has been revoked.
+		/// \[unique identifier, controller\]
+		Revoke { identifier: UniqueIdOf, author: UniqueCreatorIdOf<T> },
```
missing update

smohan-dw commented on pallets/unique/src/lib.rs:
```diff
@@ -0,0 +1,429 @@
+
+			Ok(())
+		}
+		/// Revokes a unique.
```
With an identifier-led approach, having support for updating the identifier with an updated digest is good.

smohan-dw commented on pallets/unique/src/types.rs:
```diff
@@ -0,0 +1,91 @@
+
+/// Defining the possible actions that can be taken on a stream.
+#[derive(Clone, Copy, RuntimeDebug, Decode, Encode, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum UniqueCommitActionOf {
```
Remove the digest from this. need to have a discussion on other values that are not supported atm

rohibhan committed:
Swaaped names

rohibhan commented:
> Can you rebase it on top of develop ?

rohibhan committed:
update functionality added

rohibhan committed:
added remove functionality

rohibhan committed:
removed digest

rohibhan commented:
closing PR as new branch named unique_pallet_sc is created
